packages/commons/src/auth/authenticationMiddleware.ts

```diff
@@ -27,12 +27,24 @@ export const authenticationMiddleware: (
 
     try {
       const jwtToken = jwtFromAuthHeader(req, ctx.logger);
-      const valid = await verifyJwtToken(jwtToken, config, ctx.logger);
-      if (!valid) {
-        throw unauthorizedError("Invalid token");
+      const { verified, decoded } = await verifyJwtToken(
+        jwtToken,
+        config,
+        ctx.logger
+      );
+
+      const authData: AuthData = readAuthDataFromJwtToken(
```

Comment:
if this is moved after [this check ](https://github.com/pagopa/interop-be-monorepo/pull/1454/files?diff=split&w=1#diff-dc29ecb11e35ebde8fe6b5febc76f366bd5f8b57f277f8cea45c1ec6c1a87a4bR42), and the check is done on `decoded`, the function `readAuthDataFromJwtToken` can take a decoded value that cannot be undefined, allowing to remove [optionality](https://github.com/pagopa/interop-be-monorepo/pull/1454/files?diff=split&w=1#diff-0cd91e0c754df9eaaeca52d641cbaa0b325dbb4361747e81c3dedf1f7aab58baR38) and a [second decode](https://github.com/pagopa/interop-be-monorepo/pull/1454/files?diff=split&w=1#diff-0cd91e0c754df9eaaeca52d641cbaa0b325dbb4361747e81c3dedf1f7aab58baR41)